### PR TITLE
CART-89 mercury: Update mercury to v2.0.1rc1

### PR DESCRIPTION
**Pull request by frostedcmos**
- Update to mercury v2.0.1rc1

Signed-off-by: Alexander Oganezov <alexander.a.oganezov@intel.com>

### CODE REVIEW

File: mercury.spec [hunk 1]
```diff
@@ -56,7 +59,7 @@ cmake -DMERCURY_USE_CHECKSUMS=OFF                \
       -DMERCURY_USE_SYSTEM_BOOST=ON              \
       -DMERCURY_USE_SELF_FORWARD=ON              \
       -DMERCURY_ENABLE_VERBOSE_ERROR=ON          \
-      -DBUILD_TESTING=ON                         \
+      -DBUILD_TESTING=OFF                        \
```
Comment (frostedcmos) [inline mercury.spec:62]: As per suggestion from Jerome building of mercury tests has to be disabled for now due to issues with kwsys

File: mercury.spec [hunk 2]
```diff
@@ -1,13 +1,16 @@
+%global dl_version 2.0.1rc1
+
 Name: mercury
-Version: 2.0.0
+Version: 2.0.1~rc1
 Release: 1%{?dist}
```
Comment (brianjmurrell) [inline mercury.spec:5]: Let's see if we can make this more consistent across RCs and GAs...

```suggestion
Name: mercury
Version: 2.0.1~rc1
Release: 1%{?dist}

# dl_version is version with ~ removed
%{lua:
    rpm.define("dl_version " .. string.gsub(rpm.expand("%{version}"), "~", ""))
}
```

File: Makefile [hunk 3]
```diff
@@ -1,5 +1,6 @@
 NAME       := mercury
-SRC_EXT    := bz2
+SRC_EXT    := gz
+DL_VERSION := 2.0.1rc1
```
Comment (brianjmurrell) [inline Makefile:3]: Make the following change in packaging/Makefile_packaging.mk:

```
--- a/packaging/Makefile_packaging.mk
+++ b/packaging/Makefile_packaging.mk
@@ -131,7 +131,7 @@ _topdir/SOURCES/%: % | _topdir/SOURCES/
 # At least one spec file, SLURM (sles), has a different version for the
 # download file than the version in the spec file.
 ifeq ($(DL_VERSION),)
-DL_VERSION = $(VERSION)
+DL_VERSION = $(subst ~,,$(VERSION))
 endif
 ifeq ($(DL_NAME),)
 DL_NAME = $(NAME)
```

and you can get rid of the above line.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]
I tested the suggested changes with both 2.0.0 GA and 2.0.1~rc1 and both built with changing just the `Version:`.

A move to 2.0.1 means we can get rid of some of the hard-pinning on 2.0.0 if we want now.

**Review by brianjmurrell — APPROVED** [review]
Looks good.  Hopefully the latest changes mean less change when moving from RCs to GAs and back.

**Review by jolivier23 — APPROVED** [review]

**Pull request merged**